R/assign.R

```diff
@@ -0,0 +1,473 @@
+#' tracks individual plants through time
+#'
+#' @return
+#' @export
+#'
+#' @examples
+
+# required packages -------------------------------------------------------
+library(sf)
+library(mapview) #don't actually need for function, just for checking
+library(dplyr)
+library(tidyr)
+
+# example input data ------------------------------------------------------
+# grasslandData (or exact same format), subset to a unique site, quad,
+# species
+load("./data/grasslandData.rda")
+load("./data/grasslandInventory.rda")
+# prepares the dataset to feed into the 'assign' function (the 'Assign'
+# function will do this ahead of time when the user calls it)
+sampleDat <- grasslandData[grasslandData$Site == "CO"
+                           & grasslandData$Quad == "unun_11"
+                           & grasslandData$Species == "Bouteloua gracilis",]
+# this should be a data.frame
+
+# get the appropriate grasslandInventory data for the "unun_11" quadrat,
+# to tell the 'assign' function when the quadrat was sampled
+sampleInv <- grasslandInventory[["unun_11"]]
+# this should be an integer vector
+
+# 'assign' function ---------------------------------------------------------
+
+assign <- function(sampleDat, inv, dorm, buff, buffGenet, clonal,...){
+  # arguments ---------------------------------------------------------------
+  ## double check the format of the inputs, and add additional columns required
+  dat <- st_sf(sampleDat) # data in 'grasslandData' format, must be an sf object
+    ## add columns for trackID, age, size_t+1, and recruit
+    dat$trackID <- NA
+    dat$age <- NA
+    dat$size_tplus1 <- NA
+    dat$recruit <- NA
+    dat$survives_tplus1 <- NA
+    dat$index <- c(1:nrow(dat))## assign an arbitrary, unique index number to
+    # each row in the dataset
+    dat$ghost = NA
+    dat$trackID = NA
+    dat$rametArea = NA
+  inv <- sort(sampleInv) ## integer vector of quadrat sampling years in
```

Comment:
`inv` is an argument in the function but it is immediately overwritten here. Also `sampleInv` is not defined or passed in as an argument.

Reply:
I think I fixed this? 
Line 48 is making sure that the inventory vector is in sequential order